### ISSUE TITLE
Connection closed in server but not in client

### ISSUE THREAD

**Issue by vgomez-dataworks** [dunglas/mercure#395]
Hello, i am facing problems in the connection between a dart (flutter) client and mercure.

I tryed with many libraries, even with raw connection but I always have the same problem.

When I stablish a connection with server sent events or response as stream in dart I am able to listen the messages in the topic choosen but if the client does not recibe a message after some time, the server logs a disconnection from the client, but in the client the connection is still open, so there is no way to reconnect automatically because the event is not triggered.

When the server logs the disconnection, the client stops from receibing messages.

I used mercure 0.10.4 Darwin (OS X) and the online demo.

**Comment by dunglas** [comment]
It's weird, the TCP connection should timeout at some point.

**Comment by vgomez-dataworks** [comment]
I added a timmer to ask every 30 seconds if the connection is still open, and I found that the connection is still open after 1 hour even with 30 seconds timeout in the library (in this case I am using Restio with SSE client)

**Comment by dunglas** [comment]
Could you provide a reproducer? I don't know Flutter at all but I can take a look.

**Comment by vgomez-dataworks** [comment]
Sorry for the late update.

I found that when I stop the Mercure server, the dart side marks the connection as open, so the problem is not in the Mercure side.

I made a workaround for this, I added a timer every 30 seconds in which the connection is forced to close and reconnect sending the last update id and I got a reasonable result for my use case.

Anyway I oppened a issue in the git of the client library for this particular case.
